### DISCUSSION TITLE
Treatment of Virial Data in training

### DISCUSSION THREAD

**Discussion by sivonxay** [deepmodeling/deepmd-kit#213] [Q&A] [answered]
When training a model using VASP AIMD data, should the kinetic contributions to the virials be subtracted out? 

I took a look at how it's done in https://github.com/deepmodeling/dpdata/blob/master/dpdata/vasp/outcar.py and that seems to include the kinetic contributions, since it parses the lines beginning with "in kB".

![Screen Shot 2020-04-09 at 4 59 48 PM](https://user-images.githubusercontent.com/25111118/78950414-86da9d80-7a83-11ea-95eb-60e776e443bc.png)

I have run VASP AIMD of my system at various temperatures (1000, 1500, 2000 etc.), each at a volume that yields approximately 0 external pressure (Using ISIF=2). Thus, the virial data for each of these would be similar, although their volumes are very different. Not excluding the kinetic contribution would imply that the training/model has some concept of the temperature/velocity of the training data - which is data not included in the training data.

Thanks in advance for your response,
Eric

**Comment by jameswind** [comment, marked as answer]
Hi Eric,

There are two kinds of kinetic contributions to the stress: the electronic
one and the ionic one. If we understand correctly, what you suggest
removing, with the keyword "Kinetic", is actually the electronic
contribution, which should be included and does not has the concept of
(ionic) temperature/velocity. Therefore, we think the line we used with the
keyword "in kB" should be correct.
BTW, the ionic contribution should be in the line with keyword "kinetic
pressure (ideal gas correction)". I didn't find a good reference but if you
have VASP 5.4.4, you could check the source code in force.F below the line "sum
the ionic kinetic contributions to the stress tensor".

Best,
